app/actions/authActions.js

```diff
@@ -35,38 +36,53 @@ type AccountType = ?{
   signingFunction?: Function,
   isHardwareLogin: boolean,
   isWatchOnly?: boolean,
+  hasInternetConnectivity: boolean,
 }
 
 export const ID = 'auth'
 
+export const checkForInternetConnectivity = (): Promise<boolean> =>
+  new Promise(resolve => {
+    dns.lookup('google.com', err => {
```

Comment:
Without adding any sort of external library this seemed like the least bad option

Reply:
`dns.resolve()` may be a better option here due to performance concerns as well as consistently using the network as mentioned in https://nodejs.org/api/dns.html#dns_implementation_considerations